readthedocs/embed/v3/tests/test_access.py

```diff
@@ -0,0 +1,120 @@
+from contextlib import contextmanager
+from unittest import mock
+
+import pytest
+from django.contrib.auth.models import User
+from django.test import TestCase, override_settings
+from django.urls import reverse
+from django_dynamic_fixture import get
+
+from readthedocs.builds.constants import LATEST
+from readthedocs.embed.v3.views import EmbedAPIBase
+from readthedocs.projects.constants import PRIVATE, PUBLIC
+from readthedocs.projects.models import Project
+from readthedocs.subscriptions.constants import TYPE_EMBED_API
+
+
+@override_settings(
+    USE_SUBDOMAIN=True,
+    PUBLIC_DOMAIN="readthedocs.io",
+    RTD_ALLOW_ORGAZATIONS=False,
+    RTD_DEFAULT_FEATURES={
+        TYPE_EMBED_API: 1,
+    },
+)
+@mock.patch("readthedocs.embed.v3.views.build_media_storage")
+class TestEmbedAPIV3Access(TestCase):
+    def setUp(self):
+        self.user = get(User)
+        self.project = get(
+            Project,
+            slug="docs",
+            privacy_level=PUBLIC,
+            users=[self.user],
+        )
+        self.version = self.project.versions.get(slug=LATEST)
+        self.version.privacy_level = PUBLIC
+        self.version.save()
+        self.url = (
+            reverse("embed_api_v3") + "?url=https://docs.readthedocs.io/en/latest/"
+        )
+        self.content = """
+        <html>
+            <div role=main>
+                Content
+            </div>
+        </html>
+        """
+
+    def get(self, *args, **kwargs):
+        """Wrapper around ``client.get`` to be overridden in the proxied api tests."""
+        return self.client.get(*args, **kwargs)
+
+    def _mock_open(self, content):
+        @contextmanager
+        def f(*args, **kwargs):
+            read_mock = mock.MagicMock()
+            read_mock.read.return_value = content
+            yield read_mock
+
+        return f
+
+    def _mock_storage(self, storage_mock):
+        storage_mock.open.side_effect = self._mock_open(self.content)
+
+    def test_get_content_public_version_anonymous_user(self, storage_mock):
+        self._mock_storage(storage_mock)
+        self.client.logout()
+        resp = self.get(self.url)
+        self.assertEqual(resp.status_code, 200)
```

Comment:
I think all the tests just verify status code. Maybe at least one test can also verify that the content is correctly returned :)